frontend/packages/console-dynamic-plugin-sdk/src/app/modal-support/useModal.ts

```diff
@@ -0,0 +1,9 @@
+import * as React from 'react';
+import { LaunchModal, ModalContext } from './ModalProvider';
+
+type UseModalLauncher = () => LaunchModal;
+
+export const useModal: UseModalLauncher = () => {
+  const { launchModal } = React.useContext(ModalContext);
+  return React.useMemo(() => launchModal, [launchModal]);
```

Comment:
No need for a memo here.
```suggestion
  return launchModal;
```